Parser/Services/HtmlParserService.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using Parser.Infrastructure;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Parser.Services
+{
+    public class HtmlParserService : IHtmlParserService
+    {
+
+        private readonly IEnumerable<IHtmlProvider> _providers;
+        
+        public HtmlParserService(IEnumerable<IHtmlProvider> providers)
+        {
+            _providers = providers;
+        }
+        
+        public async Task<string> GetSiteTitleAsync(string url)
+        {
+            var provider = await FindImplementation(url);
+            
+            string title = await GetTitle(await provider.GetPageHtmlAsync(url));
+            if (string.IsNullOrEmpty(title))
+            {
+                throw new ArgumentException();
```

Comment:
The idea of using your own custom exception is to avoid ambiguity. When I am reading the controller I have no idea what stands behind ArgumentException. The idea is to name the exception in a way it can be clear enough to understand on the controller level.